### PR TITLE
[AST] Make ValueDecls displayable via their "print decl ref".

### PR DESCRIPTION
**Pull request by DougGregor**

### CODE REVIEW

**Comment by DougGregor** [comment]
@swift-ci please smoke test and merge

File: include/swift/AST/Decl.h [hunk 1]
```diff
@@ -6765,6 +6766,9 @@ inline EnumElementDecl *EnumDecl::getUniqueElement(bool hasValue) const {
 std::pair<DefaultArgumentKind, Type>
 getDefaultArgumentInfo(ValueDecl *source, unsigned Index);
 
+/// Display ValueDecl subclasses.
+void simple_display(llvm::raw_ostream &out, const ValueDecl *&decl);
```
Comment (jrose-apple) [inline include/swift/AST/Decl.h]: Why the reference here?

Comment (DougGregor) [reply include/swift/AST/Decl.h]: Hmmm. The reference is pointless. I was directly specializing the more-generic version of this function. Will fix when I can tolerate touching `Decl.h`

Comment (DougGregor) [reply include/swift/AST/Decl.h]: (It's fixed now)

**Pull request merged**